### DISCUSSION TITLE
Is possible to use as event handling?

### DISCUSSION THREAD

**Discussion by nathan130200** [sebastienros/jint#1057] [Q&A] [answered]
I'm creating a game server, i can use Jint as external event handling?

Example:

Console.CancelKeyPress event, ask for a Javascript jint scripts and these scripts can modify the values, like:

```js
function OnCancelKeyPress(e){
	e.Cancel = true;
	console.log('i cancelled this event :P');
}
```

I don't have ideias how to implement or if this is possible in Jint Engine... (Dynamic Script Executing)

**Comment by ghost** [comment, marked as answer]
yes. Jint convert C# event to **add_NameEvent** for subscribe and **remove_NameEvent** for unsubscribe 
example:
C#
```csharp
public class Test()
{
    public event Action SomeEvent;
}
```

in js
```javaScript
    function Handler(){
        Log("Some happend")
    }

    var testClass = new Test()
    testClass.add_SomeEvent(Handler)
    testClass.remove_SomeEvent(Handler)
```
